### PR TITLE
Adds the plasmaman sith suit and changes a little how bundles work

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
### The Plasmaman Sith Suit
- You acquire it by rolling the sith bundle from the syndicate bundle
- The suit can't be taken off once worn, but the helmet can
- Both the suit and the helmet reduce melee damage by 60%, bullet damage by 50%, laser damage by 50%, are radiation-proof and protect well against explosives (but nothing protects enough to make you truly protected from explosives)

Thanks to @Buchse for the sprites

### Changed bundle and uplink
- I needed to add the previous changes so I've done this to allow that.
- spawn_item instead of spawning an item will actually run another proc that handles the spawning, meaning the New() can be customized on a per-uplink-entry basis
- One such interaction is with the syndicate bundle, which currently does a "is the user a plasmaman?" check, and if it is it adds the "plasmaman" condition to the list of conditions, which will then be shoved into the New() of the bundle which now contains the list of conditions to allow for different things.
- The other interaction is that the emag is now intentionally infinite and not unintentionally infinite.
- Added a testing variable for the bundle which allows you to force what bundle you get.
- Fixed a small bug where the syndicate wrapping paper costed 6 TC regardless of whether you were a cargo employee (who are supposed to get a discount to 4 TC) or not.

The PR is ready.
:cl:
 * rscadd: Added the Plasmaman Sith Suit, a durable suit that can be acquired by plasmamen who would otherwise get the sith suit from the syndicate bundle.

### CODE REVIEW

**Comment by Irradiation** [comment]
What does the suit look like?

**Comment by boy2mantwicethefam** [comment]
You can check how it looks by pressing "details" for the icon diff bot
suit
helmet
full_body + esword

**Comment by SkowronX** [comment]
The helmet should be a unique sprite too. Why not modify the visor?

**Comment by SkowronX** [comment]
In addition, the obj sprites should not have directional states and the helmet sprite is not consistent with the other plasmaman helmet sprites.

**Comment by boy2mantwicethefam** [comment]
Thanks for reminding me about the directional sprites, I forgot about that
The other item sprites are pending which is why I said it's not finished yet

**Comment by Gazathor** [comment]
>not fully protected from explosions
bro palpy survived the death star exploding you are ruining our coherent lore 0/10 you'll cost us our Disney partnership

**Comment by boy2mantwicethefam** [comment]
New menacing helmet sprite, thanks to @Buchse 
helmet

**Comment by boy2mantwicethefam** [comment]
The PR is ready for all intents and purposes

**Review by ShiftyRail — COMMENTED** [review]
This is fine as is, but I would rather have uplink bundles box be subtypes rather than dependent on a string

**Comment by boy2mantwicethefam** [comment]
One thing at a time

**Comment by yackemflam** [comment]
2 problems with this
1. This isn't considered as wizard garbs so the plasmameme can't cast his one spell
2. Why does this get a shit ton of armor in nearly everything? The robes doesn't give armor and only gives you the ability to cast spells, which you have to sacrifice to wear a space proof suit.
This literally gives a plasmaman an armor that's better than the captain's hardsuit and an esword.

**Comment by jknpj** [comment]
>Why does this get a shit ton of armor in nearly everything?

Because barry

**Comment by boy2mantwicethefam** [comment]
1. Actually the plasmaman can cast the bound object spell, it is robeless
2. Plasmaman is tankier this way. Still gets defeated by the lol stuns thing but I wanted to aim for some sort of intimidating factor, the previous idea was giving the plasmaman an occasional anti-stun.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 1]
```diff
@@ -422,3 +422,28 @@
 	base_state = "plasmamanCult_helmet"
 	desc = "A containment suit designed by the followers of Nar-Sie. It glows menacingly with unearthly flames."
 	armor = list(melee = 60, bullet = 50, laser = 30,energy = 15, bomb = 30, bio = 30, rad = 30)
+
+//Sith
+
+/obj/item/clothing/suit/space/plasmaman/sith
+	name = "Plasmaman Sith Suit"
```
Comment (Kurfursten) [inline code/modules/clothing/spacesuits/plasmamen.dm]: Sith is a proper round but the rest should not be capitalized

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 2]
```diff
@@ -422,3 +422,28 @@
...
+	item_state = "plasmaman_sith_suit"
+	desc = "A menacing armored suit that protects the wearer from harm, fit for a plasmaman. It appears to permanently seal itself once worn."
+	armor = list(melee = 60, bullet = 50, laser = 50, energy = 50, bomb = 80, bio = 100, rad = 100)
+	canremove = 0
+
+/obj/item/clothing/suit/space/plasmaman/sith/acidable()
+	return 0
+
+//Unlike the suit, the helmet can be taken off
+/obj/item/clothing/head/helmet/space/plasmaman/sith
```
Comment (Kurfursten) [inline code/modules/clothing/spacesuits/plasmamen.dm:440]: same

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 3]
```diff
@@ -1,6 +1,9 @@
-/obj/item/weapon/storage/box/syndicate/New()
+/obj/item/weapon/storage/box/syndicate/New(var/loc, var/list/conditions, var/forced_bundle)
 	..()
-	var/tagname = pickweight(list("bloodyspai" = 100, "stealth" = 100, "screwed" = 100, "guns" = 100, "murder" = 100, "freedom" = 100, "hacker" = 100, "lordsingulo" = 100, "smoothoperator" = 100, "psycho" = 100, "hotline" = 100, "ocelot" = 100, "palpatine" = 100, "anarchist" = 50, "emagsandglue" = 10, "balloon" = 10, "bangerboy" = 100, "highlander" = 100))
+	var/tagname = pickweight(list("bloodyspai" = 100, "stealth" = 100, "screwed" = 100, "guns" = 100, "murder" = 100, "freedom" = 100, "hacker" = 100, "lordsingulo" = 100, "smoothoperator" = 100, "psycho" = 100, "hotline" = 100, "ocelot" = 100, "sith" = 100, "anarchist" = 50, "emagsandglue" = 10, "balloon" = 10, "bangerboy" = 100, "highlander" = 100))
+
+	if(forced_bundle)
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: why would you put this AFTER the pickweight? you're doing a completely unnecessary list calculation if you have a forced bundle

File: code/datums/uplink_item.dm [hunk 4]
```diff
@@ -533,6 +541,12 @@ var/list/uplink_items = list()
 	item = /obj/item/weapon/storage/box/syndicate
 	cost = 14
 
+/datum/uplink_item/badass/bundle/new_uplink_item(new_item, location, user)
+	var/list/conditions = list()
+	if(isplasmaman(user))
+		conditions += "plasmaman"
```
Comment (Kurfursten) [inline code/datums/uplink_item.dm:547]: probably defines

File: code/datums/uplink_item.dm [hunk 5]
```diff
@@ -362,6 +367,9 @@ var/list/uplink_items = list()
 	item = /obj/item/weapon/card/emag
 	cost = 6
 
+/datum/uplink_item/device_tools/emag/new_uplink_item(new_item, turf/location, mob/user)
+	new new_item(location, 1) //Uplink emags are infinite
```
Comment (Kurfursten) [inline code/datums/uplink_item.dm:371]: undocumented change?

File: code/datums/uplink_item.dm (hunk 4 above)
Comment (boy2mantwicethefam) [reply code/datums/uplink_item.dm:547]: I don't think this needs defines yet

File: code/datums/uplink_item.dm (hunk 5 above)
Comment (boy2mantwicethefam) [reply code/datums/uplink_item.dm:371]: It's the fourth on the list of "changed bundle and uplink", emags used to be unintentionally infinite but they are now intentionally infinite

**Pull request merged**